plutus-tx-plugin/test/Budget/Spec.hs

```diff
@@ -155,3 +160,25 @@ compiledShow :: CompiledCode Integer
 compiledShow = $$(compile [||
       let x = -1234567890
        in showExample x ||])
+
+-- | In this example, the float-in pass cannot reduce the cost unless it allows
+-- unconditionally floating into type abstractions. Both branches are
+-- turned into type abstractions (because the `a + a` branch is not a value).
```

Comment:
That doesn't seem right? The branches aren't in a multi-occurence context so floating in should have been fine already? Same applies below. Did you observe a difference when turning it on?

Reply:
When floating into the argument of an application, it is considered multi-occ context, unless the function is either `\x -> ...` where `x` occurs once, or a builtin. If the function is a `Let` or `App` etc. then it is considered multi-occ context. Do you have a better approach?

Reply:
Aha, so the multi-occ comes from the fact that the case expression is compiled to a function application, of course. Brain fart.